### PR TITLE
License headers fixed or added using mvn license:format

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
Ah, it is failing because Travis relies in maven central only. Let me fix it.

File: .gitignore [hunk 1]
```diff
@@ -7,7 +7,16 @@ local.properties
 .idea/
 *.iml
 
+# Eclipse
```
Comment (arturdryomov) [inline .gitignore:8]: Not sure we should keep it for Studio-based project, but if it is more comfortable for you — not a problem.

File: .gitignore [hunk 2]
```diff
@@ -7,7 +7,16 @@ local.properties
...
+.classpath
+.settings
+
+
 # Gradle
 
 .gradle/
 build/
+
+# Maven
```
Comment (arturdryomov) [inline .gitignore:17]: Nit. An empty line after the header.

**Comment by arturdryomov** [comment]
These directories and files should be ignored, because they are auto-generated.
- `gradle`
- `gradlew`
- `gradlew.bat`
- `*.properties` — not in the repository, but are generated after building a project.

**Comment by arturdryomov** [comment]
@ppalaga Should I fix checkstyle warnings in your branch somehow or just merge and fix afterwards?

**Comment by arturdryomov** [comment]
Another two nits.
- Should the licence header be followed by an empty line like [there](https://github.com/hawkular/hawkular-inventory/blob/7ae4039bea8c82e96ff4516ab75178dd7eb2e27a/api/src/main/java/org/hawkular/inventory/api/InventoryException.java#L17)?
- Should I use `mvn -s .travis.maven.settings.xml verify` locally as well?

File: .gitignore (hunk 1 above)
Comment (ppalaga) [inline .gitignore:8]: Not sure what Studio-based should imply here... Is this not a usual Gradle project that can be handled by any Gradle-compatible tool incl. Eclipse?

Comment (arturdryomov) [inline .gitignore:8]: As far as I know Eclipse doesn’t support it — at least yet. You can keep, not a big deal.

File: .gitignore (hunk 2 above)
Comment (ppalaga) [inline .gitignore:17]: You mean the an empty line after the header is missing? I'd say that there is an empty line too much after all other section markers :)

Comment (arturdryomov) [inline .gitignore:17]: That’s why it is nit!

**Comment by ppalaga** [comment]
> These directories and files should be ignored, because they are auto-generated.
> gradle gradlew gradlew.bat *.properties — not in the repository, but are generated after building a project.

Hm... some of them were in git even before my changes. Let me try to remove them.

> Should the licence header be followed by an empty line?

If `mvn license:format` did without an empty line, I'd say that's correct.

> Should I use mvn -s .travis.maven.settings.xml verify locally as well?

Yes, you can, but when thinking of it again, it downloads a lot of maven stuff that is not necessary for the license plugin  and checkstyle. Let me try to find a more sparse form of the mvn command.

**Comment by arturdryomov** [comment]
> Hm... some of them were in git even before my changes. Let me try to remove them.

I meant ignored by checking plugins.

**Comment by ppalaga** [comment]
> > Hm... some of them were in git even before my changes. Let me try to remove them.
> 
> I meant ignored by checking plugins.

OK, I excluded them from license checks https://github.com/ppalaga/android-client/commit/edac4f7b5b197dca948d13b9160211391e36b36e I hope the list is complete

**Comment by ppalaga** [comment]
The last commit is a result of bulk Source > Format operation in Eclipse. Not sure if this will suit Android Studio. Feel free to criticize or propose something else. Anyway, Checkstyle is passing like that. https://github.com/hawkular/android-client/commit/13c52e0538d73b6cf39521756a43ebd4587fdcba

**Comment by ppalaga** [comment]
> Should I use mvn -s .travis.maven.settings.xml verify locally as well?

I changed to `mvn license:check checkstyle:check` However, I am not 100% sure it saved any bandwith.

**Comment by arturdryomov** [comment]
marvelous

@ppalaga Is the `.travis.maven.settings.xml` really necessary though? Seems like it just adds repositories and the `hawkular-parent` is available at Central. Isn’t it better to move these declarations to the parent POM directly?

**Comment by ppalaga** [comment]
@ming13 yes, `.travis.maven.settings.xml` makes sense, because `hawkular-parent` reaches maven central within 24hrs from the release to JBoss Nexus and I typically upgrade parent in consuming projects just after releasing to nexus.

**Comment by arturdryomov** [comment]
@ppalaga, thanks for your work!

**Pull request merged**